examples/visualization/ui_events.py

```diff
@@ -0,0 +1,100 @@
+"""
+.. _ex-ui-events:
+
+======================================
+Using the event system to link figures
+======================================
+
+Many of MNE-Python's figures are interactive. For example, you can select channels or
+scroll through time. The event system allows you to link figures together so that
+interacting with one figure will simultaneously update another figure.
+
+In this example, we'll be looking at linking two topomap plots, such that selecting the
+time in one will also update the time in the other, as well as hooking our own
+custom plot into MNE-Python's event system.
```

Comment:
FYI another long-awaited feature I think has been linking together to `Brain` plots, maybe also two `raw` plots. It doesn't have to be in this PR but adding these at some point would be awesome! Thinking ahead, I wonder if we want a `tutorials/visualization/ui_events.py` or something. If you agree you could `git mv` this file over there (new directory) to start. `examples/visualization/publication_figure.py` could probably also move over there I think (with a new `redirect` added to `doc/conf.py` since it has existed already).

Reply:
During this PR I discovered that [`mne.viz.link_brains`](https://mne.tools/stable/generated/mne.viz.link_brains.html) is a thing.

What would be the use case for linking two `raw` plots? Anyway, the event system allows you to link anything to anything, as long as the plotting functions are modified to produce and consume UI events. I've been having a lot of fun playing around with this already :)

Ah, good idea to move `publication_figure.py` as well.  I wanted the `ui_events` example to be a tutorial initially, but I couldn't find a good place to put it.

Reply:
> During this PR I discovered that [mne.viz.link_brains](https://mne.tools/stable/generated/mne.viz.link_brains.html) is a thing.

Ideally I think `link_brains` would be `@deprecated` or at least `@legacy` as part of this overall effort (in this PR or a follow-up), or whichever one makes this work natively/easily for `Brain` objects. Whenever that's done, adding a brain example to the tutorial would be good!

> What would be the use case for linking two raw plots? 

A hyperscanning paradigm would be one use case. But maybe we don't have the data to show it. We could fake it by `.pick('eeg')` and `.pick('meg')` on `sample` to show something simultaneous. But if it gets too convoluted I don't think it's 100% necessary.

Thinking further, a really cool one would actually be to link `epochs.plot` to a `raw.plot` where clicking in a given epoch goes back to the same chunk of data for raw. :)

> Ah, good idea to move publication_figure.py as well. I wanted the ui_events example to be a tutorial initially, but I couldn't find a good place to put it.

Yeah I think adding a new folder `tutorials/visualization` makes sense for stuff that's not typically restricted to raw/epochs/evoked. Since this is designed to span all -- and sometimes cross between as above -- these levels I think it makes sense to have a new folder.

Reply:
... and if we `@legacy` the `link_brains` rather than deprecate it, ideally it would use these mechanisms under the hood rather than using whatever code path it does currently.